test/test-setup.js

```diff
@@ -0,0 +1,100 @@
+"use strict";
+
+const test = require("sdk/test");
+const tabs = require("sdk/tabs");
+const {ActivityStreams} = require("lib/ActivityStreams");
+const httpd = require("./lib/httpd");
+const {doGetFile} = require("./lib/utils");
+
+const PORT = 8099;
+const path = "/dummy-activitystreams.html";
+const url = `http://localhost:${PORT}${path}`;
+const otherUrl = `http://localhost:${PORT}/dummy-other.html`;
+
+let srv;
+let app;
+let openTabs;
+
+function beforeEach() {
+  srv = httpd.startServerAsync(PORT, null, doGetFile("test/resources"));
+  app = new ActivityStreams({pageURL: url});
+  openTabs = [];
+
+  function onOpen(tab) {
+    openTabs.push(tab);
+  }
+  tabs.on("open", onOpen);
+}
+
+function afterEachPromise() {
+  return new Promise((resolve, reject) => {
+    app.unload();
+    Promise.all([
+      ...openTabs.map(tab => new Promise(resolve => tab.close(resolve))),
+      new Promise(resolve => srv.stop(resolve))
+    ]).then(resolve).catch(reject);
+  });
+}
+
+function getFirstWorker() {
+  return app.workers.values().next().value;
+}
+
+exports["test load worker"] = function*(assert) {
+  beforeEach();
+  yield new Promise(resolve => {
+    function onReady(tab) {
+      if (tab.url === url) {
+        assert.equal(app.workers.size, 1, "Worker is loaded");
+      }
+      resolve();
+    }
+    tabs.open({url, onReady});
+  });
```

Comment:
Maybe a better test would be to have no tabs, test the number of workers (zero), open a new tab, test (one worker).

Same applies for other tests